### DISCUSSION TITLE
graphql resolving and many attemps reduce our speed to boost applications

### DISCUSSION THREAD

**Discussion by loveyandex** [vendurehq/vendure#1169] [General]
i have a suggestion to automate many works about reoslving in plugin design
it is not good really to use graphql

**Comment by michaelbromley** [comment]
If you are suggesting that Vendure moves away from GraphQL, this is not going to happen. If you want to use Vendure, you need to learn GraphQL. If you don't want to learn GraphQL, you need to find a different platform to build on.

I am open to suggestions on how to improve the docs though for those not yet familiar with GraphQL.

**Reply by loveyandex** [reply]
thanks great man u are right 
this decision is base of vendure and can not change with costs

i prefer study your modeling to create a new exactly equal vendure 
maybe i have more question but thank u alot  u are here

**Comment by loveyandex** [comment]
can u explain deffrent with facets and Collections

i understand faces is same attibues of a product 
and collections are same category for products
we can use this way ?

**Comment by loveyandex** [comment]
here there is a paradox
u said in collections this ::
        `Collections allow you to group ProductVariants together by various criteria. A typical use of Collections is to create a `hierarchical category `tree which can be used in a navigation menu in your storefront.`


and also said in facet doc :
category

![image](https://user-images.githubusercontent.com/53381292/137886288-5f666f95-bf7e-48a1-90ad-cdeac4119d6d.png)

**Reply by michaelbromley** [reply]
Yes, collections _can_ be used to represent "categories" for navigation. That is not the only use though. Perhaps the demo data is a bit confusing because it has a Facet named "category". I suggest you play around with it to get a feel for how it works.

**Reply by heiko-r** [reply]
I think it makes sense to have categories both as a facet and as a collection. You use facets to add information to your products. Facets are great for searching and filtering, but they have no hierarchy. Collections instead are much less flexible than facets, but they have hierarchy, so you can use them to build your shop navigation.
So I think the best way is to 'tag' every product with one or more 'category' facets and then create your collections based on these facets.
